CheckLevelUnlocked.gd

```diff
@@ -0,0 +1,23 @@
+extends Node
+
+#the global singleton dealing with the stage engine 
```

Comment:
I don't love singletons as a pattern, is there a different way to expose these properties without having global state? Alternatively, are there fields which can be moved elsewhere or made private so that we have a minimal amount of global state?

Reply:
I can definitely minimize parts of it, but it would require saving the scene. I don't know how to check if a level is complete without a global variable due to being unable to get a value from an unloaded scene. 

help